packages/contracts-bedrock/STYLE_GUIDE.md

```diff
@@ -45,6 +45,18 @@ We also have the following custom tags:
 
 - Event parameters should NOT be prefixed with an underscore.
 
+#### Immutable variables
+
+Immutable variables:
+
+- should be in `SCREAMING_SNAKE_CASE`
+- should be `internal`
+- should have a hand written getter function
+
+This approach clearly indicates to the developer that the value is immutable, without exposing
+the non-standard casing to the interface. It also ensure that we don’t need to break the ABIs if
+we switch between values being in storage and immutable.
```

Comment:
Not sure if this approach is set in stone, but another idea is moving this concern out into one or more `OptimismLens` style contracts that have all relevant getter methods. That way concerns are separated: the implementations don't need to concern themselves with getter-noise and backwards compatibility, and devs building on top always enter through the lens contract to fetch data

Reply:
Interesting. I do agree that writing all these custom getters is quite a pain. 

We could perhaps do this in the `SystemConfig` contract.

For now I wouldn't say that it's 'set in stone', but is part of a future tech debt paydown we should plan to do.